pinot-core/src/main/java/org/apache/pinot/core/transport/QueryRouter.java

```diff
@@ -174,7 +174,7 @@ public void shutDown() {
     _serverChannels.shutDown();
   }
 
-  void receiveDataTable(ServerRoutingInstance serverRoutingInstance, DataTable dataTable, int responseSize,
+  long receiveDataTable(ServerRoutingInstance serverRoutingInstance, DataTable dataTable, int responseSize,
```

Comment:
From clean API perspective, doesn't seem intuitive that why it should return requestId.

If the caller already has DataTable, then can it pass it to this function by extracting it from dataTable.getMetadata() and this function not having to do it again. 

Ok with the change though if it is not feasible.

Reply:
Fixed.